File: MyLanguageWeb/src/main/resources/static/js/classes/dto/entity/workout/types/favourite/favourite_customer_collection_response_dto.js
import {
    CustomerCollectionResponseDTO
} from "../../../customer_collection/response/customer_collection_response_dto.js";

export class FavouriteCustomerCollectionResponseDTO {
    #numberOfWorkouts;
    #customerCollection;

    constructor(json) {
        if (json) {
            this.#numberOfWorkouts = json["number_of_workouts"];

            let customerCollection = json["customer_collection"];
            if (customerCollection) {
                this.#customerCollection = new CustomerCollectionResponseDTO(customerCollection);
            }
        }
    }

    getNumberOfWorkouts() {
        return this.#numberOfWorkouts;
    }

    getCustomerCollection() {
        return this.#customerCollection;
    }
}

File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/dto/entity/workout/types/favourite/customer_collection/FavouriteCustomerCollectionResponseDTO.java
package ru.herooo.mylanguageweb.dto.entity.workout.types.favourite.customer_collection;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import ru.herooo.mylanguageweb.controllers.json.l.LongDeserializer;
import ru.herooo.mylanguageweb.controllers.json.l.LongSerializer;
import ru.herooo.mylanguageweb.dto.entity.customercollection.response.CustomerCollectionResponseDTO;

public class FavouriteCustomerCollectionResponseDTO {
    @JsonProperty("customer_collection")
    private CustomerCollectionResponseDTO customerCollection;

    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("number_of_workouts")
    private long numberOfWorkouts;

    public CustomerCollectionResponseDTO getCustomerCollection() {
        return customerCollection;
    }

    public void setCustomerCollection(CustomerCollectionResponseDTO customerCollection) {
        this.customerCollection = customerCollection;
    }

    public long getNumberOfWorkouts() {
        return numberOfWorkouts;
    }

    public void setNumberOfWorkouts(long numberOfWorkouts) {
        this.numberOfWorkouts = numberOfWorkouts;
    }
}
